File: dtc.frontend/src/dtos/Player/UpdatePlayerDto.ts
import z from "zod";

export const updatePlayerDtoSchema = z.object({
  firstName: z
    .string({
      message: "Der Vorname ist ein Pflichtfeld.",
    })
    .min(1, "Der Vorname ist ein Pflichtfeld.")
    .max(50, "Der Vorname darf maximal 50 Zeichen lang sein."),

  lastName: z
    .string({
      message: "Der Nachname ist ein Pflichtfeld.",
    })
    .min(1, "Der Nachname ist ein Pflichtfeld.")
    .max(50, "Der Nachname darf maximal 50 Zeichen lang sein."),

  nickname: z
    .string()
    .max(50, "Der Spitzname darf maximal 50 Zeichen lang sein.")
    .nullable(),
});

export type UpdatePlayerDto = z.infer<typeof updatePlayerDtoSchema>;

File: DTC.Api/Dtos/Player/UpdatePlayerDto.cs
using System.ComponentModel.DataAnnotations;

namespace DTC.Api.Dtos.Player
{
    public class UpdatePlayerDto
    {
        [Required(ErrorMessage = "Der Vorname ist ein Pflichtfeld.")]
        [StringLength(50, ErrorMessage = "Der Vorname darf maximal 50 Zeichen lang sein.")]
        public string FirstName { get; set; } = string.Empty;

        [Required(ErrorMessage = "Der Nachname ist ein Pflichtfeld.")]
        [StringLength(50, ErrorMessage = "Der Nachname darf maximal 50 Zeichen lang sein.")]
        public string LastName { get; set; } = string.Empty;

        [StringLength(50, ErrorMessage = "Der Spitzname darf maximal 50 Zeichen lang sein.")]
        public string? Nickname { get; set; }
    }
}
